flair/models/clustering.py

```diff
@@ -0,0 +1,112 @@
+import pickle
+from pathlib import Path
+from typing import Union, List
+
+import torch
+from sklearn.metrics import normalized_mutual_info_score, silhouette_samples
+from tqdm import tqdm
+from collections import OrderedDict
+
+from sklearn.base import ClusterMixin, BaseEstimator
+from sklearn.model_selection import train_test_split
+
+from flair.data import Corpus, Sentence
+from flair.datasets import DataLoader
+from flair.embeddings import DocumentEmbeddings
+
+import logging
+
+log = logging.getLogger("flair")
+
+
+class ClusteringModel:
+    """
+    A wrapper class for the sklearn clustering models. With this class clustering with the library 'flair' can be done.
+    """
+
+    def __init__(
+        self, model: Union[ClusterMixin, BaseEstimator], corpus: Corpus, label_type: str, embeddings: DocumentEmbeddings
+    ):
+        """
+          :param model: the clustering algortihm from sklearn this wrapper will use.
+          :param corpus: the flair corpus this wrapper will use for clustering.
+          :param label_type: the label from the sentence will be used for the evaluation.
+          :param embeddings: the flair DocumentEmbedding this wrapper uses to calculate a vector for each sentence.
+        """
+
+        self.model = model
+        self.corpus = corpus
+        self.label_type = label_type
+        self.embeddings = embeddings
+
+    def fit(self, **kwargs):
+        """
+        Trains the model.
+        """
+        X, y = self._convert_dataset()
+        X_train, X_test, y_train, y_test = train_test_split(X, y)
+        self.model.fit(X_train, **kwargs)
+
+    def predict(self, sentences: Union[List[Sentence], Sentence]):
+        """
+        Predict labels given a list of sentences and returns the respective class indices.
+        """
+        X = [sentence.embedding.cpu().detach().numpy() for sentence in sentences]
+        predict = self.model.predict(X)
+
+        for idx, sentence in enumerate(sentences):
+            sentence.set_label("cluster", str(predict[idx]))
+
+        return predict
+
+    def save(self, model_file: Union[str, Path]):
+        """
+        Saves current model.
+        """
+        dump = pickle.dumps(self.model)
+        torch.save(dump, str(model_file), pickle_protocol=4)
+
+        log.info("Saved model to: " + str(model_file))
+
+    def load(self, model_file: Union[str, Path]):
```

Comment:
Should be a static method, since it initializes the object. E.g. the following code snippet should be the way to load and use a trained clustering model: 

```python
    # load saved clustering model
    model = ClusteringModel.load(model_file="clustering_model.pt")
    
    # make example sentence
    sentence = Sentence('Getting error in manage categories - not found for attribute "navigation _ column"')
    
    # predict for sentence
    model.predict(sentence)
    
    # print sentence with prediction
    print(sentence)
```